addons/web/static/src/js/core/dom.js

```diff
@@ -221,7 +221,12 @@ return {
     renderButton: function (options) {
         var params = options.attrs || {};
         params.type = params.type || 'button';
-        params.class = 'btn btn-' + (options.size || 'sm') + ' ' + (params.class || 'btn-default');
+        var classes = 'btn-default';
+        if (params.class) {
+            classes = params.class.replace(/\boe_highlight\b/g, 'btn-primary')
```

Comment:
add comment specifying that this is backward compatibility?

Reply:
Not sure. Depends if we want to get rid of `oe_hightlight` in views / use only this class / use a new class / keep using it or `btn-primary` (as we discussed regarding the last commit of this PR)

Reply:
i'd be happy if people stop using it... one concept one keyword/classname

Reply:
Yes, but what keyword? :) `btn-primary` depends on bootstrap. `oe_hightlight`... is a crappy name :)

Reply:
whatever as soon as there is only one

Reply:
@aab-odoo I changed the PR following your comments, except for this one. Should I merge the PR without the last commit and leave it for some more framework-team-thinking ?

Reply:
What does @JKE-be think about it? For me it's ok...

Reply:
@JKE-be What do you think about commit: https://github.com/odoo/odoo/pull/23115/commits/511395219fd848df8c3e4f2518d5f71aaa792541 ? The JS code would still replace them automatically with `btn-primary` but we might want to use only one class in views for consistency (same goes for `oe_link`/`btn-link`). Is it worth breaking clients' xpaths ?